onadata/apps/api/models/odk_token.py

```diff
@@ -0,0 +1,51 @@
+"""
+ODK token model class
+"""
+import binascii
+import os
+
+from django.conf import settings
+from django.db import models
+from django.db.models.signals import post_save
+
+from django_digest.models import (_persist_partial_digests,
+                                  _prepare_partial_digests)
+
+AUTH_USER_MODEL = getattr(settings, 'AUTH_USER_MODEL', 'auth.User')
+ODK_TOKEN_LENGTH = getattr(settings, 'ODK_TOKEN_LENGTH', 7)
+
+
+class ODKToken(models.Model):
+    """
+    ODK Token class
+    """
+    key = models.CharField(max_length=40, primary_key=True)
+    user = models.OneToOneField(
+        AUTH_USER_MODEL, related_name='odk_token', on_delete=models.CASCADE)
+    created = models.DateTimeField(auto_now_add=True)
```

Comment:
I think we may need to have a `status` column and maintain one and only one active token per user. I am hoping it is useful for troubleshooting and auditing purposes. That way, we never really delete a token but make it inactive.

Reply:
@ukanga In this case will regenerating tokens not be supported ? Or will the user be able to do both ?

Reply:
Should we update the last time the user used the token? like a `last_login` field or should we use the `user.last_login` on successful authentication?

Reply:
Utilizing the `user.last_login` seems like the most ideal way in my opinion. I've implemented the change on the `DigestAuthentication` class to update the `user.last_login` on successful authentication.